internal/controller/apiextensions/claim/reconciler.go

```diff
@@ -332,6 +376,14 @@ func (r *Reconciler) Reconcile(ctx context.Context, req reconcile.Request) (reco
 				record.Event(cm, event.Warning(reasonDelete, err))
 				return reconcile.Result{}, err
 			}
+
+		}
+
+		if err := r.claim.UnpublishConnection(ctx, cm, nil); err != nil {
+			log.Debug(errDeleteCDs, "error", err)
+			err = errors.Wrap(err, errDeleteCDs)
+			record.Event(cm, event.Warning(reasonDelete, err))
+			return reconcile.Result{}, err
```

Comment:
Maybe worth a small comment explaining that we're deleting the propagated connection secret, just because there's no corresponding "publish connection" call in this method, unlike others? I'm on the fence though - might be overkill.